data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisService.java

```diff
@@ -0,0 +1,150 @@
+            throw new IllegalStateException("Lease Coordination table should be provided!");
+        }
+        KinesisLeaseConfig kinesisLeaseConfig =
+                kinesisLeaseConfigSupplier.getKinesisExtensionLeaseConfig().get();
+        this.tableName = kinesisLeaseConfig.getLeaseCoordinationTable().getTableName();
+        this.kclMetricsNamespaceName = this.tableName;
+        this.dynamoDbClient = kinesisClientFactory.buildDynamoDBClient(kinesisLeaseConfig.getLeaseCoordinationTable().getAwsRegion());
+        this.kinesisClient = kinesisClientFactory.buildKinesisAsyncClient();
+        this.cloudWatchClient = kinesisClientFactory.buildCloudWatchAsyncClient(kinesisLeaseConfig.getLeaseCoordinationTable().getAwsRegion());
+        this.pipelineName = pipelineDescription.getPipelineName();
+        this.applicationName = pipelineName;
+        this.executorService = Executors.newFixedThreadPool(1);
+    }
+
+    public void start(final Buffer<Record<Event>> buffer) {
+        if (buffer == null) {
+            throw new IllegalStateException("Buffer provided is null");
+        }
+
+        if (sourceConfig.getStreams() == null || sourceConfig.getStreams().isEmpty()) {
+            throw new IllegalStateException("Streams are empty!");
+        }
+
+        scheduler = getScheduler(buffer);
+        executorService.execute(scheduler);
+    }
+
+    public void shutDown() {
+        LOG.info("Stop request received for Kinesis Source");
+
+        Future<Boolean> gracefulShutdownFuture = scheduler.startGracefulShutdown();
+        LOG.info("Waiting up to {} seconds for shutdown to complete.", GRACEFUL_SHUTDOWN_WAIT_INTERVAL_SECONDS);
+        try {
+            gracefulShutdownFuture.get(GRACEFUL_SHUTDOWN_WAIT_INTERVAL_SECONDS, TimeUnit.SECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException ex) {
+            LOG.error("Exception while executing kinesis consumer graceful shutdown, doing force shutdown", ex);
+            scheduler.shutdown();
+        }
+        LOG.info("Completed, shutting down now.");
+    }
+
+    public Scheduler getScheduler(final Buffer<Record<Event>> buffer) {
+        if (scheduler == null) {
+            return createScheduler(buffer);
+        }
+        return scheduler;
+    }
+
+    public Scheduler createScheduler(final Buffer<Record<Event>> buffer) {
+        final ShardRecordProcessorFactory processorFactory = new KinesisShardRecordProcessorFactory(
+                buffer, sourceConfig, acknowledgementSetManager, pluginMetrics, pluginFactory);
+
+        ConfigsBuilder configsBuilder =
+                new ConfigsBuilder(
+                        new KinesisMultiStreamTracker(kinesisClient, sourceConfig, applicationName),
+                        applicationName, kinesisClient, dynamoDbClient, cloudWatchClient,
+                        new WorkerIdentifierGenerator().generate(), processorFactory
```

Comment:
Let's inject this into the constructor.

Also, make `WorkIdentifierGenerator` an interface. Then name the current class into `HostNameWorkerIdentifierGenerator`.

Reply:
Thanks @dlvenable. I have fixed this.